pkg/controller/webhookbootstrap/controller.go

```diff
@@ -315,7 +315,7 @@ func (c *controller) certificateRequiresIssuance(ctx context.Context, log logr.L
 	}
 
 	// validate the common name is correct
-	expectedCN := pki.CommonNameForCertificate(crt)
+	expectedCN := crt.Spec.CommonName
 	if expectedCN != cert.Subject.CommonName {
```

Comment:
If `crt.Spec.CommonName` is not set, and `cert.Subject.CommonName` *is* set, we are not correctly handling it here.

Reply:
Is that true?
```
expectedCN := ""
cert.Subject.CommonName := "foo"
expectedCN != cert.Subject.CommonName == `true` 
```

Reply:
Right, but aren't we meant to be permissive in this case? So if expectedCN is "" (because it wasn't specified on the Certificate), and the x509 cert's CN field *is* set to `foo`, this should only trigger a re-issue *iff* `foo` is not specified as part of `certificate.spec.dnsNames`.

In this iteration of the code, we'll trigger a re-issue if the expected CN and the actual CN are different, no matter what.

Reply:
I see, this makes sense